src/components/ArrayInput.tsx

```diff
@@ -0,0 +1,70 @@
+/* @jsx jsx */
+import { jsx } from '@emotion/core';
+import * as React from 'react';
+
+import { Box, Button, Flex, Text } from '@stoplight/ui-kit';
+
+import { fieldName, IFormtronControl } from '..';
+
+export const ArrayInput: React.FunctionComponent<IFormtronControl> = ({
+  id,
+  value = [],
+  schema,
+  onChange,
+  fieldComponents,
+  selection,
+}) => {
+  // Make shallow copy
+  const items = [...value];
+  const update = (index: number, val: any) => {
+    items.splice(index, 1, val);
+    return items;
+  };
+  const insert = (index: number) => {
+    items.splice(index, 0, defaultValue);
+    return items;
+  };
+  const append = (val: any) => {
+    items.push(val);
+    return items;
+  };
+  const remove = (index: number) => {
+    items.splice(index, 1);
+    return items;
+  };
+  const defaultValue = schema.default;
+  const Widget = fieldComponents[fieldName(schema.items)];
```

Comment:
New line

Reply:
Are you serious? Or just doing this because you saw my tweet?
https://twitter.com/wmhilton/status/1088945595757281281

Reply:
I didn't know you posted a tweet about this argument.

I believe it helps the readability — as it helps to separate a function from another.

```ts
  const update = (index: number, val: any) => {
    items.splice(index, 1, val);
    return items;
  };
  const insert = (index: number) => {
    items.splice(index, 0, defaultValue);
    return items;
  };
  const append = (val: any) => {
    items.push(val);
    return items;
  };
  const remove = (index: number) => {
    items.splice(index, 1);
    return items;
  };
```

```ts
  const update = (index: number, val: any) => {
    items.splice(index, 1, val);
    return items;
  };

  const insert = (index: number) => {
    items.splice(index, 0, defaultValue);
    return items;
  };

  const append = (val: any) => {
    items.push(val);
    return items;
  };

  const remove = (index: number) => {
    items.splice(index, 1);
    return items;
  };
```

I believe the latter is easier to read.

Reply:
Sure, whatever. But if Prettier has a config for this lets just set it. I don't want to think about whitespace formatting.

Reply:
I would have made them one-liners but prettier made them 4-liners.

Reply:
how about this:
```js
  const update = (index: number, val: any) => (items.splice(index, 1, val), items);
  const insert = (index: number) => (items.splice(index, 0, defaultValue), items);
  const append = (val: any) => (items.push(val), items);
  const remove = (index: number) => (items.splice(index, 1), items);
```
:trollface:

Reply:
You're awakening my old skills in C++. If I recall correctly, the comma operator simply returns what's on the right side of it — transforming the two statements in an expression, isn't it?

Reply:
It's horribly confusing to me, because it looks like it's returning some kind of tuple `(a, b)`.

Reply:
But that's about the right amount of LOC that I think those four functions deserve. 😆

Reply:
How about this?
```js

  // Make shallow copy
  const items = [...value];
  const defaultValue = schema.default;
  const Widget = fieldComponents[fieldName(schema.items)];

  const splice = (start: number, deleteCount: number, ...vals: any[]) => {
    items.splice(start, deleteCount, ...vals);
    return items;
  };

  const update = (index: number, val: any) => splice(index, 1, val);

  const insert = (index: number) => splice(index, 0, defaultValue);

  const append = () => splice(items.length, 0, defaultValue);

  const remove = (index: number) => splice(index, 1);

  return (
```

Reply:
See https://github.com/stoplightio/formtron/pull/17#discussion_r251785036 :trollface: